File: classTypes4/java/src/Mamifero.java
// extiende de AnimalBase para heredar atributos y el método mover()
public abstract class Mamifero extends AnimalBase {
    // Atributo especifico de mamiferos
    private int gestacionMeses;

    public Mamifero(String nombreCientifico, double pesoKg, int gestacionMeses) {
        // llama al constructor de AnimalBase
        super(nombreCientifico, pesoKg); 
        this.gestacionMeses = gestacionMeses;
    }
    
    //todos los mamíferos amamantan a sus crías
    public void amamantar() {
        System.out.println(getNombreCientifico() + " esta amamantando a sus crias");
    }
    
    // hereda el método abstracto emitirSonido() de AnimalBase, comprometiendo a las clases hijas a implementarlo.
}

File: classTypes4/c#/Mamifero.cs
// extiende de animalbase para heredar atributos y el metodo mover()
public abstract class Mamifero : AnimalBase {
    // atributo especifico de mamiferos
    private int gestacionMeses;

    public Mamifero(string nombreCientifico, double pesoKg, int gestacionMeses) 
        // llama al constructor de animalbase
        : base(nombreCientifico, pesoKg){
        this.gestacionMeses = gestacionMeses;
    }
    
    // todos los mamiferos amamantan a sus crias
    public void amamantar(){
        Console.WriteLine(getNombreCientifico().ToLower() + " esta amamantando a sus crias");
    }
    
    // hereda el metodo abstracto emitirsonido() de animalbase.
}